graph/appender/unused_service.go

```diff
@@ -47,25 +48,30 @@ func addUnusedNodes(trees *[]*tree.ServiceNode, namespaceName string, deployment
 	}
 }
 
-func buildStaticNodeList(namespaceName string, deployments *v1beta1.DeploymentList) []*tree.ServiceNode {
+func buildStaticNodeList(namespaceName string, pods *v1.PodList) []*tree.ServiceNode {
 	nonTrafficList := make([]*tree.ServiceNode, 0)
+	resolvedServices := make(map[string]bool)
 	appLabel := config.Get().ServiceFilterLabelName
 	versionLabel := config.Get().VersionFilterLabelName
 	identityDomain := config.Get().Products.Istio.IstioIdentityDomain
-	for _, deployment := range deployments.Items {
-		app, ok := deployment.GetObjectMeta().GetLabels()[appLabel]
+	for _, pod := range pods.Items {
+		app, ok := pod.GetObjectMeta().GetLabels()[appLabel]
 		if !ok {
-			log.Warningf("Deployment %s has not a proper app label [%s]", deployment.Name, appLabel)
+			log.Warningf("Pod %s has not a proper app label [%s]", pod.Name, appLabel)
 			continue
 		}
-		version, ok := deployment.GetObjectMeta().GetLabels()[versionLabel]
+		version, ok := pod.GetObjectMeta().GetLabels()[versionLabel]
 		if !ok {
-			log.Warningf("Deployment %s has not a proper version label [%s]", deployment.Name, versionLabel)
+			log.Warningf("Pod %s has not a proper version label [%s]", pod.Name, versionLabel)
 			continue
 		}
-		staticNode := tree.NewServiceNode(fmt.Sprintf("%s.%s.%s", app, namespaceName, identityDomain), version)
-		staticNode.Metadata = map[string]interface{}{"rate": 0.0, "isUnused": "true"}
-		nonTrafficList = append(nonTrafficList, &staticNode)
+		srvId := fmt.Sprintf("%s.%s.%s.%s", app, namespaceName, identityDomain, version)
+		if _, ok := resolvedServices[srvId]; !ok {
```

Comment:
Ok, this check is introduced as now, using pods, they can be several pods with same app x label match.